### ISSUE TITLE
`spago test` runs script in package's directory, not workspace directory

### ISSUE THREAD

**Issue by JordanMartinez** [purescript/spago#984]
See https://github.com/JordanMartinez/spago-next-bug for a reproducible example, which surprised me.

I think there are three responses to this issue.:
1. always run `spago test` in the workspace directory
1. always run `spago test` in the package directory
1. always run `spago test` in the directory in the current working directory

Most of the time, if I'm running a test, I execute the `spago test` command when my current working directory is the workspace directory.

**Comment by f-f** [comment]
Yeah this is a known behaviour, and we somewhat thought it through, though I now believe that our assumptions were likely wrong: we settled on the idea of "`spago test -p somepackage` runs in the directory of the package" when integrating the new spago with the `registry-dev` codebase, and I think we thought that made sense because there was only one package, and in the process of splitting it in many sub-packages we didn't want to make "global" assumptions. That is, the idea of depending on some "global" path didn't feel right when deep in some focused and self-contained package.

However, the reality of things is that in a monorepo packages can't stand on their own: you can't go in the directory of a subpackage and run `spago test`: it's not going to work because spago doesn't have any information about the workspace configuration, which is found only at the workspace root.
So then the root of the monorepo is the only place where one can run `spago test`, and it doesn't make sense to relocate the cwd to a different place, forcing the user to deal with a program that can be executed from two distinct locations.

This is all a longwinded way of saying that I think that option 1 above is the only sensible path forward (and in fact `spago run` already behaves like that)

cc @thomashoneyman

**Comment by thomashoneyman** [comment]
I think commands in a workspace should run from the workspace root, so 👍

**Comment by JordanMartinez** [comment]
I started working on this but because the `spaghetto` dir uses `spago`, I couldn't build the project:
```
Downloading dependencies...
Building...
[ 1 of 10] Compiling Spago.Generated.BuildInfo
[ 2 of 10] Compiling Spago.BuildInfo
[1/1 TypesDoNotUnify] src/Spago/BuildInfo.purs:77:23

  77  currentSpagoVersion = BuildInfo.buildInfo.packages."spago-bin"
                            ^^^^^^^^^^^^^^^^^^^

  Could not match type
    Record
  with type
    Array
  while trying to match type { "spago-bin" :: String
                             | t0
                             }
    with type Array
                { name :: String
                , version :: String
                }
  while matching label packages
  while checking that expression buildInfo
    has type { packages :: { "spago-bin" :: String
                           | t0
                           }
             | t1
             }
  while checking type of property accessor buildInfo.packages
  while checking type of property accessor (buildInfo.packages)."spago-bin"
  in value declaration currentSpagoVersion
  where t0 is an unknown type
        t1 is an unknown type
```

**Comment by f-f** [comment]
@JordanMartinez ah yes, I fixed the format for the generated build info in a breaking way, so you'd need to upgrade. I just published `0.93.11` which should work

**Comment by JordanMartinez** [comment]
> @JordanMartinez ah yes, I fixed the format for the generated build info in a breaking way, so you'd need to upgrade. I just published `0.93.11` which should work

How often do you think that module will change in a breaking way? I find it weird that code other than the actual source code gets compiled in the first place, but I also don't know your reasons behind doing things that way. Is there somewhere where I can read about this design decision?

Otherwise, I've submitted a PR for this.

**Comment by f-f** [comment]
Thanks!

The reasoning for generating build info to include in the build is in https://github.com/purescript/spago/issues/599.

I don't think there's any other way to achieve this (Cabal in Haskell does the same to make the version field available to the package that is being compiled, which is where I based my implementation from).

I expect things to break all the time until we declare the new Spago stable, at which point we'll still be able to change this feature, but we'll have to do it in a backwards compatible way.
